src/frontend/src/session/cursor_manager.rs

```diff
@@ -471,16 +483,36 @@ impl SubscriptionCursor {
                     ans.push(row);
                 }
                 None => {
-                    break;
+                    if cur == 0 {
+                        // It's only blocked when there's no data
+                        // This method will only be called once, either to trigger a timeout or to get the return value in the next loop via `next_row`.
+                        session
+                            .env
+                            .hummock_snapshot_manager()
+                            .wait_table_change_log_notification(
```

Comment:
When `wait_table_change_log_notification` timeout triggers, we are relying on the if condition in L504 to hit to exit the while loop. This is somehow strange and dangerous. I susggest we make the following changes:
1. Make `wait_table_change_log_notification` to wait for notification without a timeout.
2. Call `tokio::time::timeout(timeout_seconds, wait_table_change_log_notification)` here.
3. When 2 returns `time::error::Elapsed`, break the loop directly.